--- a/build/toolchain/win/toolchain.gni
+++ b/build/toolchain/win/toolchain.gni
@@ -891,7 +891,7 @@
 
     sys_include_flags = "${win_toolchain_data.include_flags_imsvc}"
     if (use_lld) {
-      sys_lib_flags = "-libpath:$_clang_lib_dir " +
+      sys_lib_flags = "\"-libpath:$_clang_lib_dir\" " +
                       "${win_toolchain_data.libpath_lldlink_flags}"
     }
 
@@ -911,7 +911,7 @@
 
     sys_include_flags = "${rust_host_win_toolchain_data.include_flags_imsvc}"
     if (use_lld) {
-      sys_lib_flags = "-libpath:$_clang_lib_dir " +
+      sys_lib_flags = "\"-libpath:$_clang_lib_dir\" " +
                       "${rust_host_win_toolchain_data.libpath_lldlink_flags}"
     }
 
@@ -931,7 +931,7 @@
 
     sys_include_flags = "${win_toolchain_data.include_flags_imsvc}"
     if (use_lld) {
-      sys_lib_flags = "-libpath:$_clang_lib_dir " +
+      sys_lib_flags = "\"-libpath:$_clang_lib_dir\" " +
                       "${win_toolchain_data.libpath_lldlink_flags}"
     }
 
@@ -951,7 +951,7 @@
 
     sys_include_flags = "${rust_host_win_toolchain_data.include_flags_imsvc}"
     if (use_lld) {
-      sys_lib_flags = "-libpath:$_clang_lib_dir " +
+      sys_lib_flags = "\"-libpath:$_clang_lib_dir\" " +
                       "${rust_host_win_toolchain_data.libpath_lldlink_flags}"
     }
 
